### ISSUE TITLE
GetVideoMediaStreamInfosAsync not Working 

### ISSUE THREAD

**Issue by Rajneek** [Tyrrrz/YoutubeExplode#282]
As per below code System throw exception  "Video [rv9qC2mMRjo] is unavailable."

  YoutubeClient client = new YoutubeClient();
  MediaStreamInfoSet  streamInfoSet = await client.GetVideoMediaStreamInfosAsync("rv9qC2mMRjo");

So  GetVideoMediaStreamInfosAsync not working on 90% videos.

**Comment by Henry84** [comment]
I have the same problem.
It come from YoutubeClient.Video -> GetVideoInfoDicAsync() -> 
`// If video ID is not set - throw
            if (result.GetValueOrDefault("video_id").IsNullOrWhiteSpace())
                throw new VideoUnavailableException(videoId, $"Video [{videoId}] is unavailable.");`

Seems like video_id is not in get_video_info response.
I've commented it and still working.

**Comment by valentin1628** [comment]
Hi ,
same issue ! 

thanks for your answer Henry but sorry i'm noob...

i don't understand your answer :( 

my example : 
`var id = YoutubeClient.ParseVideoId(url);
            

            var client = new YoutubeClient();
            var converter = new YoutubeConverter(client);


            var video = await client.GetVideoAsync(id); //my app freeze here and no return...
           


            var streamInfoSet = await client.GetVideoMediaStreamInfosAsync(id);

            // Select one of the streams, e.g. highest quality muxed stream
            var streamInfo = streamInfoSet.Audio.WithHighestBitrate();`

Many thanks for your help !

**Comment by Tyrrrz** [comment]
Duplicate of #281

**Comment by GlennIM** [comment]
Per the documentation on the code page, I'm trying to call the function below but the class doesn't have the WithHighestBitrate() function available.

// ...or highest bitrate audio stream
// var streamInfo = streamInfoSet.Audio.WithHighestBitrate();

So I am using one of the audio streams streamInfo[1].Url and using the Url parameter that returns a long Url with a player in it which is VERY nice. I am taking that Url and putting it in a Chromium Embedded Framework browser and it plays well.  Questions... Does that Url parameter with player ever expire? Also, do you know of any way to control the audio player playback using keyboard keys, or a way in .NET to use keyboard keys to control playback instead of the Gui?

**Comment by GlennIM** [comment]
Per the documentation on the code page, I'm trying to call the function below but the class doesn't have the WithHighestBitrate() function available.

// ...or highest bitrate audio stream
// var streamInfo = streamInfoSet.Audio.WithHighestBitrate();

So I am using one of the audio streams streamInfo[1].Url and using the Url parameter that returns a long Url with a player in it which is VERY nice. I am taking that Url and putting it in a Chromium Embedded Framework browser and it plays well.  Questions... Does that Url parameter with player ever expire? Also, do you know of any way to control the audio player playback using keyboard keys, or a way in .NET to use keyboard keys to control playback instead of the Gui?

Thanks in advance for the great lib!

**Comment by Tyrrrz** [comment]
Yeah, the URLs do expire. You can check `mediaStreamInfoSet.ValidUntil` to know when, but it's pretty much always a few hours, if I remember correctly.
